File: MySqlTestInObjectPascal/relationship.pas
unit Relationship;

{$mode objfpc}{$H+}

interface

uses
  Classes, SysUtils;

type
  TRelationship = class
  private
    { private declarations }

    // The id field of the Relationship class.
    idField: Integer;
    // The name field of the Relationship class.
    nameField: String;
    // The isDeleted field of the Relationship class.
    isDeletedField: Boolean;
  public
    { public declarations }

    { Initializes a new instance of the Relationship class. }
    constructor Create();
    { Initializes a new instance of the Relationship class. }
    constructor Create(id: Integer; name: String; isdeleted: Boolean);
    { Gets the id. }
    function getId(): Integer;
    { Sets the id. }
    procedure setId(id: Integer);
    { Gets the name. }
    function getName(): String;
    { Sets the name. }
    procedure setName(name: String);
    { Gets the isDeleted. }
    function getIsDeleted(): Boolean;
    { Sets the isDeleted. }
    procedure setIsDeleted(isdeleted: Boolean);
  end;

implementation

{ Initializes a new instance of the Relationship class. }
constructor TRelationship.Create();
begin

end;

{ Initializes a new instance of the Relationship class. }
constructor TRelationship.Create(id: Integer; name: String; isdeleted: Boolean);
begin
     idField := id;
     nameField := name;
     isDeletedField := isdeleted;
end;

{ Gets the id. }
function TRelationship.getId(): Integer;
begin
     getId := idField;
end;

{ Sets the id. }
procedure TRelationship.setId(id: Integer);
begin
     idField := id;
end;

{ Gets the name. }
function TRelationship.getName(): String;
begin
     getName := nameField;
end;

{ Sets the name. }
procedure TRelationship.setName(name: String);
begin
     nameField := name;
end;

{ Gets the isDeleted. }
function TRelationship.getIsDeleted(): Boolean;
begin
     getIsDeleted := isDeletedField;
end;

{ Sets the isDeleted. }
procedure TRelationship.setIsDeleted(isdeleted: Boolean);
begin
     isDeletedField := isdeleted;
end;

end.

File: MsSqlTestInVisualBasic/MsSqlTest/Relationship.vb
' Base class for Relationship.
Class Relationship

#Region "Fields"

    ' The id field of the Relationship class.
    Private idField As Integer

    ' The name field of the Relationship class.
    Private nameField As String

    ' The isDeleted field of the Relationship class.
    Private isDeletedField As Boolean

#End Region

#Region "Properties"

    ' Gets or sets the idField.
    Property Id() As Integer
        Get
            Return idField
        End Get
        Set(ByVal value As Integer)
            idField = value
        End Set
    End Property

    ' Gets or sets the nameField.
    Property Name() As String
        Get
            Return nameField
        End Get
        Set(ByVal value As String)
            nameField = value
        End Set
    End Property

    ' Gets or sets the isDeletedfield.
    Property IsDeleted() As Boolean
        Get
            Return isDeletedField
        End Get
        Set(ByVal value As Boolean)
            isDeletedField = value
        End Set
    End Property

#End Region

#Region "Constructors"

    ' Initializes a new instance of the Relationship class.
    Public Sub New()
    End Sub

    ' Initializes a new instance of the Relationship class.
    Public Sub New(ByVal id As Integer, ByVal name As String, ByVal isdeleted As Boolean)
        Me.idField = id
        Me.nameField = name
        Me.isDeletedField = isdeleted
    End Sub

#End Region

#Region "Methods"

#End Region

End Class
